datafusion/functions-nested/src/map.rs

```diff
@@ -61,14 +62,46 @@ fn make_map_batch(args: &[ColumnarValue]) -> datafusion_common::Result<ColumnarV
 
     let data_type = args[0].data_type();
     let can_evaluate_to_const = can_evaluate_to_const(args);
-    let key = get_first_array_ref(&args[0])?;
-    let value = get_first_array_ref(&args[1])?;
-    make_map_batch_internal(key, value, can_evaluate_to_const, data_type)
+
+    // check the keys array is unique
+    let keys = get_first_array_ref(&args[0])?;
+    let key_array = keys.as_ref();
+
+    match &args[0] {
+        ColumnarValue::Array(_) => {
+            for i in 0..key_array.len() {
+                let row_keys = key_array.slice(i, 1);
+                let row_keys = first_array_for_list(row_keys.as_ref());
```

Comment:
Can we use `datafusion_common::utils::list_to_arrays::<O>(keys)` here ?
Also any specific reason for making this a inner fn ? why not move it out ? For better readability